streams/src/main/java/org/apache/kafka/streams/processor/internals/PartitionGroup.java

```diff
@@ -49,44 +46,46 @@ public TopicPartition partition() {
             return queue.partition();
         }
 
-        public RecordQueue queue() {
+        RecordQueue queue() {
             return queue;
         }
     }
 
     // since task is thread-safe, we do not need to synchronize on local variables
     private int totalBuffered;
 
-    public PartitionGroup(Map<TopicPartition, RecordQueue> partitionQueues, TimestampExtractor timestampExtractor) {
-        this.queuesByTime = new PriorityQueue<>(partitionQueues.size(), new Comparator<RecordQueue>() {
+    PartitionGroup(final Map<TopicPartition, RecordQueue> partitionQueues) {
+        queuesByTime = new PriorityQueue<>(partitionQueues.size(), new Comparator<RecordQueue>() {
 
             @Override
-            public int compare(RecordQueue queue1, RecordQueue queue2) {
-                long time1 = queue1.timestamp();
-                long time2 = queue2.timestamp();
-
-                if (time1 < time2) return -1;
-                if (time1 > time2) return 1;
+            public int compare(final RecordQueue queue1, final RecordQueue queue2) {
+                final long time1 = queue1.timestamp();
```

Comment:
Interesting, so you have seen cases where there were bugs because local variables or method parameters were not final? I don't remember that ever happening, so interested in your experience.

Unlike fields, final in method params and local variables make no difference at runtime, don't affect the memory model and are used in a much smaller scope. In Java 8, they even introduced effectively final local variables so that one can use them in lambdas and inner classes without having to add the `final` keyword. So, the readability point is arguable (and the Java language designers seem to agree).